### ISSUE TITLE
Describing query params 

### ISSUE THREAD

**Issue by romanoaugusto88** [Teamwork/kommentaar#36]
I'm trying to generate documentation for query params like that:

```go
package controller

const (
	GetQueryParamsMinPageSize     = 10
	GetQueryParamsMaxPageSize     = 100
	GetQueryParamsDefaultPageSize = 30
)

type GetQueryParams struct {
	// PageSize sets the number of items in a page.
	//
	// minimum: $ref: GetQueryParamsMinPageSize
	// maximum: $ref: GetQueryParamsMaxPageSize
	// default: $ref: GetQueryParamsDefaultPageSize
	PageSize int `query:"pageSize"`
}

type GetResponse struct{}

// GET /foo/{id} tag
// One-line description.
//
// A more detailed multi-line description.
//
// Query: $ref: GetQueryParams
// Response 200: $ref: GetResponse
```

The `minimum`, `maximum` and `default` are being parsed as description and the `$ref` are not being interpolated.
Am I doing something wrong or is it not supported?

**Comment by Carpetsmoker** [comment]
On 27 August 2018 09:28:11 BST, Romano Augusto <notifications@github.com> wrote:
>I'm trying to generate documentation for query params like that:
>
>```go
>package controller
>
>const (
>	GetQueryParamsMinPageSize     = 10
>	GetQueryParamsMaxPageSize     = 100
>	GetQueryParamsDefaultPageSize = 30
>)
>
>type GetQueryParams struct {
>	// PageSize sets the number of items in a page.
>	//
>	// minimum: $ref: GetQueryParamsMinPageSize
>	// maximum: $ref: GetQueryParamsMaxPageSize
>	// default: $ref: GetQueryParamsDefaultPageSize
>	PageSize int `query:"pageSize"`
>}
>
>type GetResponse struct{}
>
>// GET /foo/{id} tag
>// One-line description.
>//
>// A more detailed multi-line description.
>//
>// Query: $ref: GetQueryParams
>// Response 200: $ref: GetResponse
>```
>
>The `minimum`, `maximum` and `default` are being parsed as description
>and the `$ref` are not being interpolated.
>Am I doing something wrong or is it not supported?

You always need to use a $ref to a struct, support for inline query params was removed some times ago, as it wasn't used a lot and made the code a lot more complex (should update the docs). 

You can use the same struct to bind the params to in echo BTW

**Comment by romanoaugusto88** [comment]
What kind of problems did you have before?

In Projects, there are loads of query parameters that need documentation, it would be very nice to have it.
Also, the `$ref` couldn't be any valid Go type to avoid doc comments being out of date with the code?

**Comment by romanoaugusto88** [comment]
I'm failing miserably on that, do you have an example?

```go
package controller

type MyInt64 struct {
	int64
}

var PageSizeMin = MyInt64{5}

type GetQueryParams struct {
	// PageSize sets the number of items in a page.
	//
	// minimum: $ref: PageSizeMin
	PageSize int `query:"pageSize"`
}

type GetResponse struct{}

// GET /foo/{id} tag
// One-line description.
//
// A more detailed multi-line description.
//
// Query: $ref: GetQueryParams
// Response 200: $ref: GetResponse
```

**Comment by Carpetsmoker** [comment]
Now that I am no longer on my phone, I see that I misunderstood your question entirely >_< I thought you were talking about referencing query parameters, but you're talking about additional properties for fields.

Why did you expect:

	// minimum: $ref: PageSizeMin

to work? Is that documented like that somewhere?

The syntax is `{property}` or `{property: value}`, for example:

```
type GetQueryParams struct {
	// PageSize sets the number of items in a page
	// {default: 30, range: 10-100}.
	PageSize int `query:"pageSize"`
}
```

Or, if you prefer:

```
type GetQueryParams struct {
	// PageSize sets the number of items in a page
	// {default: 30} {range: 10-100}.
	PageSize int `query:"pageSize"`
}
```

Both should work identical.

You can't reference stuff in these sort of properties. I don't know if we need to support that?

I'll write more complete docs on this later today.

**Comment by romanoaugusto88** [comment]
Thanks :+1: 

Now I'm trying to figure out a way to share the description across common params. Looks like it doesn't work with embedded structs, is it a bug or feature? :smile: 

```go
type CommonQueryParams struct {
	// FieldTasks common description.
	//
	// {enum: name priority status description}
	FieldTasks *[]string `query:"fields[tasks]"`
}

type GetQueryParams struct {
	CommonQueryParams
	// {default: 10, range: 10-100, required}
	PageSize int64 `query:"pageSize"`
}
```
```bash
$ kommentaar .
schema is nil for field fields[tasks] in controller.GetQueryParams
```

If I remove the tag, it works though :thinking:

**Comment by Carpetsmoker** [comment]
That was a bug; embedded structs didn't work correctly with query parameters as it would use the `json` tag instead of `query`. That should be fixed with #40 now; I added your example as a test case, and things should work as expected if you update.
